### ISSUE TITLE
Convert ARX forecaster to the correct format

### ISSUE THREAD

**Issue by dajmcdon** [cmu-delphi/epipredict#3]
The proper forecaster signature should be:

```
prob_arx <- function(x, y, geo_value, time_value, args = prob_arx_args())
```

All args not in the `epi_` data frame should be in an args list.

**Comment by brookslogan** [comment]
I think we may want to consider instead
```
function(df, args = prob_arx_args())
```
or something mirroring `fable`, whatever its interface is.

As:
- More commonly than not, users will probably will want (x,y,geo_value,time_value) in a data frame format to pass to a fitter function like `lm`, `rq`, etc.  If they are given `df` as an input, this is less trouble.
- The code to call a forecaster / list of forecasters seems easier:
```
fetch_latest_data_from_my_favorite_api() %>% prob_arx(my_args)
as_of(archive, forecast_date) %>% prob_arx(my_args)
snapshot = as_of(archive, forecast_date)
lapply(forecasters_with_args, function(forecaster_with_args) {
  forecaster_with_args$forecaster(snapshot, forecaster_with_args$args)
})
```
as opposed to something with `do.call` or `tidy_eval` or things along those lines.  Although I'm not sure if taking `df` as an argument would make things easier or harder when using `epi[x]_slide`; I forget if slider expects functions to take in the columns or the df.

**Comment by dajmcdon** [comment]
Addressed in #10
